pkgs/development/python-modules/amaranth-boards/default.nix

```diff
@@ -19,7 +19,9 @@ buildPythonPackage rec {
     owner = "amaranth-lang";
     repo = "amaranth-boards";
     rev = "aba2300dc83216523e1c98fdb22471cb4bac5027";
-    hash = "sha256-IMhNMkOf6dg4Djyi39TDTpU0S3TQM1+e6TBoyiITaos=";
+    # these files change depending on git branch status
+    postFetch = "rm -f $out/.git_archival.txt $out/.gitattributes";
```

Comment:
I found the context https://github.com/NixOS/nixpkgs/issues/84312

This is soooo bad.

Reply:
Did not realize this was known to nixpkgs, thanks for linking. Yeah it's a really unfortunate mis-feature.

Reply:
I'm actually pretty sad about the removal of `.git_archival.txt` since it means that anyone using tarballs gets an archive which cannot be installed without additional knowledge (of what the version number is supposed to be). pdm-backend doesn't support this one way or another, but if it did I would have preferred to keep it...

Reply:
I don’t know if there were improvements to be made to the actual implementation, but the version being not fully dependent on the git hash is strange to me to say the least.

Reply:
I am fairly confident there is a way to implement `.git_archival.txt` which would not change the tarball iff your versioning strategy (tag assignment) does not result in retroactively changing the version for a normal git checkout (which is also very bad for obvious reasons).

I am also fairly confident that export-subst is too crude of a mechanism to do that, so I do sympathize with both the Nix folks in particular, and with anyone complaining about reproducibility in general. I have just also had unpleasant conversations with people who reasonably expect tarballs to be buildable as-is, so it's being caught between a rock and a hard place.

Reply:
fyi yes you can soundly do this, with the invariant just changing to "you don't tag parents of commits someone put in nixpkgs" which is a lot looser of an invariant i think. you want `%(describe:abbrev=10)`. it will only catch annotated tags.

Reply:
Thanks! That would interact really poorly with our existing git history in Amaranth since all of the tags are not annotated. Can I safely replace unannotated tags with annotated ones?

Reply:
I think it should be perfectly sound to do to future ones and only then swap the substitution spec. Also there's a tags=true option on that descriptor if you want to also consider unannotated tags. and there's a filter you can use to which tags it considers. see PRETTY FORMATS in git-log(1).

I think annotating existing tags is probably safe as well. I expect it is likely safe for our code as well. 

I've not looked at the code but you may have to adjust the stuff consuming it because it will give you v1.0-50-abcdef0123 instead of just tag names on the current commit; in the case of it being a tag however it will just give you the tag itself.

Reply:
This will require a patch to `pdm-backend` which neither me nor Frost Ming seem to feel like doing, but it's very useful to have this for future reference; thanks!